src/library.js

```diff
@@ -1238,13 +1238,33 @@ mergeInto(LibraryManager.library, {
     return _strptime(buf, format, tm); // no locale support yet
   },
 
+  // Base Emscripten EH error class
+#if TRANSPILE_TO_ES5
+  // FIXME: manually transpile due to https://github.com/google/closure-compiler/issues/1138
+  $EmscriptenEH__docs: '/** @constructor */',
+  $EmscriptenEH: 'function EmscriptenEH() {}\n' +
+                 'EmscriptenEH.prototype = Error.prototype;',
+#else
+  $EmscriptenEH: 'class EmscriptenEH extends Error {}',
```

Comment:
Presumably inheriting from `Error` (which IIRC trigger the creation and tracking of a backtrace under the hood) has some performance cost.   Should we avoid doing this in release builds by default?

Reply:
Inheritance from Error should always be done for custom errors - it's what makes it provide useful backtraces, and allows to reason about the thrown object. Many libraries expect it to be a subclass of Error, with `instanceof` working as expected, properties where they should be and so on.

As for the backtraces, IIRC, the cost is negligible until you actually symbolise the error stack by printing the error or manually accessing the `stack` property - only then engines walk it and turn from internal stack to JS representation.

Reply:
IIUC the cost of adding backtraces is non-zero, and these setjmp/longjmp (and some uses of C++ exceptions) are really an internal implementation detail of emscripten and should be escape.   Keeping the default / status quo the same seems like the safest bet here.